### ISSUE TITLE
Named/Lored items don't stack properly

### ISSUE THREAD

**Issue by RedDevel2** [CivMC/Civ#510]
Items that have data since the 1.20 update no longer stack properly. They often have slight differences in the data on them (Why)
Could have something to do with the SAH Item meta converter hack no longer working properly?

This mainly affects Essence, Fossils, and compacted items.
Some of these items no longer work in the factories related to #495

**Comment by Protonull** [comment]
Right, just went about investigating this and we have an annoying problem. When [this issue](https://github.com/CivClassic/CivModCore/issues/88) was opened (on 1.16.5), setting an item's display name via `itemMeta.displayName(Component.text("Hello, World!"))` would yield the following NBT:
```
{
    "Name": '{"text":"Hello, World!"}'
}
```
Whereas, `itemMeta.setDisplayName("Hello, World!")` would yield this NBT:
```
{
    "Name": '{"text":"","extra":[{"text":"Hello, World!"}]}'
}
```

However, it seems that Minecraft has done some optimisations because components in NBT are now (in 1.20.4) looking a lot like their pre-1.13 selves, where `itemMeta.displayName(Component.text("Hello, World!"))` will yield the following NBT:
```
{
    "Name": 'Hello, World!'
}
```
And `itemMeta.setDisplayName("Hello, World!")` yields this NBT:
```
{
    "Name": '{"text":"","extra":["Hello, World!"]}'
}
```

This seems like an easy fix, right? Just do `Component.empty().append(displayName)`, except that produces this NBT:
```
{
    "Name": '{"extra":["Hello, World!"],"text":""}'
}
```

Despite these being identical JSON objects, they are not identical strings... and they are stored on items as serialised strings. I'm not sure it's possible to fix this issue. It may be worth hard-upgrading items to 1.20.4 components instead.

**Comment by okx-code** [comment]
Not relevant anymore

**Comment by Protonull** [comment]
> Not relevant anymore

I do think this'll likely come back to bite you sooner or later. For example, bastion ingredients (like Casings and Gearboxes) are still defined by item-meta lore, rather than by your custom-item api, so if Mojang changes how components are stored again then the problem will arise again. It's just not a problem *right now*.
